packages/ove-service-auth/README.md

```diff
@@ -0,0 +1,191 @@
+# OVE Base Proxy
+
+This is a generic base proxy that supports various configurations, provided either via environment variables or 
+configuration files.
+
+## Configurations
+
+### Request Proxy
+
+The request proxy supports two configurations: a global or service-based proxy. The service proxy supports basic
+load balancing.
+
+The proxy mechanism can be enabled by setting the **SERVICE_PROXY_ENABLED** environment variable, the default is **True**.
+
+In proxy mode the service maps all requests based on a path matching strategy:
+
+```python
+r'/(?P<service>.+)/(?P<path>.*)'
+```
+- the **service** part is missing in global proxy mode. The service should match the name configured in the 
+**services.json** file.
+- **path** is used in the forwarded request 
+
+#### Global proxy
+
+In **global proxy** mode all requests are forwarded to the provided url. The url is configured using the 
+**SERVICE_GLOBAL_PROXY_URL** environment variable. 
+
+**Example:** The proxy is running on **localhost:8080** and service-abc responds to requests on 
+**localhost:9090/search/<query>**. We would like to proxy all these requests and map **localhost:8080** -> 
+**localhost:9090/search/** therefore the **SERVICE_GLOBAL_PROXY_URL** will be set to **localhost:9090/search/**.
+
+#### Service proxy
+
+In **service proxy** mode, the request are mapped in a similar way, but the mapping rules are described by the 
+**services.json** config file. Example:
+
+```json
+{
+  "ove": [
+    "http://localhost:3004/",
+    "http://localhost:3005/"
+  ]
+}
+```
+
+In this case, the proxy is running on **localhost:8080** and two ove services are running on **localhost:3004** 
+and **localhost:3005**. In case of a request to **locahost:8080/ove/** this will be forwarded to either 
+**localhost:3004** or **localhost:3005** by using a very simple load balancing strategy.
+
+**Note**: This proxy service assumes that all the **state synchronisation** is performed by the services.
+
+#### Important note on proxies
+
+The proxy mapping rules are overridden by the rest service mappings. In case authentication and locking are enabled
+the proxy will serve these routes instead of proxying the requests. Currently all the auth operations are performed 
+under **auth/login** and **auth/lock** routes.
+
+### Authentication
+
+This feature allows users to authenticate with standard username and password. The authentication is performed using 
+a standard jwt protocol. The user config, with username and passwords are described in the **config/users.json** file:
```

Comment:
This is ambiguous about whether there can be multiple users ('username' is singular, but 'passwords' is plural)